engine/src/execution_graph/logic_controllers/CacheMachine.h

```diff
@@ -192,11 +285,26 @@ class WaitingQueue {
 		return this->finished.load(std::memory_order_seq_cst);
 	}
 
+
+
+	void wait_for_count(int count){
+
+		std::unique_lock<std::mutex> lock(mutex_);
+		condition_variable_.wait(lock, [&, this] () {
+			std::cout<<"message queue size is "<<this->processed<<std::endl;
+			return count <= this->processed;
```

Comment:
so this is an interesting isituation. It should really be == and if its ever > we should through an exception. not sure what you think

Reply:
I agree on both counts, but this needs to be on our branch, because this function is not used here